app/models/manageiq/providers/openstack/base_metrics_capture.rb

```diff
@@ -92,9 +93,17 @@ def list_meters(filter)
   def meter_names(metric_capture_module)
     @meter_names ||= metric_capture_module::COUNTER_NAMES
   end
+  
+  included do
+    cache_with_timeout(:counters_by_vm, 30.seconds) { Hash.new }
```

Comment:
If these counter definitions don't change often we should cache them longer than 30 seconds to get the most benefit out of the cache.  We cache the vmware counter definitions for 1 hour for example.